org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java

```diff
@@ -651,6 +657,41 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
 			format(ent);
 	}
 
+	/**
+	 * Format the diff entries by filtering out the noise from the given delta filter pattern
+	 * The filter acts only for files that have MODIFY change Type.
+	 * If there are no changes detected, we will remove the diff entry.
+	 * @param entries
+	 * @param deltaFilterPattern
+	 * @throws IOException
+	 */
+
+	public void format(List<? extends DiffEntry> entries, Pattern deltaFilterPattern) throws IOException {
+		if (deltaFilterPattern == null) {
+			format(entries);
+			return;
+		}
+		Iterator<? extends DiffEntry> diIterator = entries.iterator();
+		while (diIterator.hasNext()) {
+			DiffEntry diffEntry = diIterator.next();
+			if (MODIFY.equals(diffEntry.changeType)) {
+				FormatResult res = createFormatResult(diffEntry);
+				String aContent = new String(res.a.content);
+				String bContent = new String(res.b.content);
+				aContent = deltaFilterPattern.matcher(aContent).replaceAll(EMPTY_STRING);
+				bContent = deltaFilterPattern.matcher(bContent).replaceAll(EMPTY_STRING);
+				if (!aContent.equals(bContent)) {
+					format(res.header, res.a, res.b);
+				} else {
+					diIterator.remove();
+				}
```

Comment:
No need for the curly braces around single lines

Reply:
oh yeah, I keep forgetting to not add those braces. style guide :(.

Reply:
changed it